s3tests_boto3/functional/test_iam.py

```diff
@@ -1795,3 +1796,266 @@ def test_copy_object_allow_get_src_put_dest():
     eq(etag, response['ETag'])
     response = client.delete_user_policy(PolicyName='AllowGetPolicy', UserName=get_main_user_id())
     eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Allow and Deny Get Bucket Versioning API using IAM policy for self')
+@attr(assertion='succeeds')
+@attr('user-policy')
+@attr('test_of_iam')
+def test_allow_deny_get_bucket_versioning_iam_policy_self():
+    client = get_iam_client()
+    s3_client_iam = get_iam_s3client()
+    bucket = get_new_bucket(client=s3_client_iam)
+    s3_client_iam.put_object(Bucket=bucket, Key='iam1buk1obj1', Body='bar')
+    response = s3_client_iam.get_bucket_versioning(Bucket=bucket)
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    deny_get_versioning_policy = json.dumps(
+        {"Version": "2012-10-17",
+         "Statement": {
+             "Effect": "Deny",
+             "Action": ["s3:GetBucketVersioning"],
+             "Resource": f"arn:aws:s3:::{bucket}"
+         }
+         }
+    )
+    response = client.put_user_policy(PolicyDocument=deny_get_versioning_policy,
+                                      PolicyName='deny_policy_versioning',
+                                      UserName=get_iam_user_id())
+    eq(response['ResponseMetadata']['HTTPStatusCode'], 200)
+    response = s3_client_iam.get_bucket_versioning(Bucket=bucket)
```

Comment:
Remove this line
```suggestion

```

Reply:
this is needed to check that getbucketversioning is denied as put in line 1825

Reply:
ok